db-service/lib/deep-queries.js

```diff
@@ -114,6 +150,24 @@ const _calculateExpandColumns = (target, data, expandColumns = [], elementMap =
       expandColumns.push(expandColumn)
     }
 
+    // for deep update check if is recursiv and has looping association --> would result in cqn4sql loop
+    if(target.name === compositions[compName].target){
+      for(const compName2 in compositions){
+        if(compName2 === compName) continue
+        const innerCompDefinition = compositions[compName2]
+        const innerCompositions = innerCompDefinition?._target?.compositions
+        if(!innerCompositions || target.name === innerCompDefinition.target) continue
+        for(const innerCompName in innerCompositions){          
+          if(innerCompositions[innerCompName].target === target.name){
+            throw new Error(
```

Comment:
Rather then throwing an `Error`. This functionality is supposed to be supported. As we have the Bookshop genre test to make sure recursive `compositions` work. The only difference is that it uses a self recursive `composition`.